src/config/scrivitoSites.ts

```diff
@@ -31,27 +46,56 @@ export function siteForUrl(
   }
 
   const baseAppUrl = getBaseAppUrl()
+  const language = languageForUrl(url, baseAppUrl)
+  const languageSite = language
+    ? appWebsites().and('_language', 'equals', language).first()
+    : undefined
+  const languageSiteId = languageSite?.siteId()
+
+  if (!languageSiteId) return findSiteForUrl(url)
+
+  return { baseUrl: `${baseAppUrl}/${language}`, siteId: languageSiteId }
+}
+
+function languageForUrl(url: string, baseAppUrl?: string) {
   if (!baseAppUrl) return
 
   const regex = new RegExp(`^${baseAppUrl}\\/(?<lang>[a-z]{2})([?/]|$)`)
-  const language = regex.exec(url)?.groups?.lang
-  if (!language) return
+  return regex.exec(url)?.groups?.lang
+}
 
-  const languageSite = allWebsites()
-    .and('_language', 'equals', language)
-    .first()
-  if (!languageSite) return
+function findSiteForUrl(url: string) {
+  const sites = allWebsites()
+    .toArray()
+    .filter((site) =>
+      baseUrlsFor(site).some((baseUrl) => url.startsWith(baseUrl)),
+    )
+  if (sites.length !== 1) return
 
-  const languageSiteId = languageSite.siteId()
-  if (!languageSiteId) return
+  const site = sites[0]
+  const siteId = site?.siteId()
+  if (!site || !siteId) return
 
-  return { baseUrl: `${baseAppUrl}/${language}`, siteId: languageSiteId }
+  const baseUrl = baseUrlsFor(site)[0]
+  if (!baseUrl) return
+
+  return { baseUrl, siteId }
+}
```

Comment:
Do I understand this correctly, that `baseUrl` can be potentially have many URLs and that this code will "normalize" the `baseUrl` to the first in the list?

Reply:
If we don't need the normalization we could potentially simplify this code.

Reply:
`findSiteForUrl` is the fallback for sites that don't match any of the app sites/versions.
* Yes, such a site may have more than one `baseUrl` (for example, prod and staging).
* Yes, we use the first URL as the primary URL, in order to avoid a mismatch between `baseUrlForSite` and `siteForUrl`.

According to this
>  If the site has the domain all to itself (like in the above example), it is sufficient for siteForUrl to return this ID. Otherwise, if two or more sites share a domain (like in the example below), siteForUrl should return the site’s ID as well as its base URL.

we should always return the base URL, because we don't have control over the content.

Under some circumstances Scrivito will complain if the base URLs of the two callbacks don't match. Unfortunately the docs don't touch these cases, so I'm not sure about potential risks. I could try it out, but even if it works now, will it continue to do so in the future?

Reply:
If I'm not mistaken this function can be simplified to:

```ts
function findSiteForUrl(url: string) {
  for (const site of appWebsites().toArray()) {
    const baseUrls = baseUrlsFor(site)
    const siteId = site.siteId()
    if (!siteId) continue

    if (baseUrls.some((baseUrl) => url.startsWith(baseUrl))) {
      return { baseUrl: baseUrls[0], siteId }
    }
  }
}
```

The only difference I see to the previous implementation is, that if two or more websites do have a matching `baseUrlsFor` the previous algorithm would return `undefined`, where as my proposal would return the first match it can find.

Reply:
> the previous algorithm would return undefined

This was intentional to clearly indicate that after copy/pasting a new website the UI cannot handle the source and destination, and a base URL must be changed.
However, I'll remove the check for now and simplify.